### PR TITLE
plugins/lsp: modernize implem of language-servers

### PR DESCRIPTION
**Pull request by GaetanLepage**

### CODE REVIEW

File: plugins/lsp/language-servers/ltex-settings.nix [hunk 1]
```diff
@@ -212,52 +215,50 @@ with lib;
...
+    This setting is a multi-scope setting. See the documentation for details.
+    This setting supports external files. See the documentation for details.
+    If this list is very large, performance may suffer.
+
+    Example:
+    ```nix
+      {
+        "en-US" = [ ":/path/to/externalFile.txt" ];
+      }
+    ```
```
Comment (MattSturgeon) [inline plugins/lsp/language-servers/ltex-settings.nix:258]: Can we promote this to be a "real" example?

**Review by MattSturgeon — COMMENTED** [review]
Looked through and found no issues.

I spotted one manually added markdown example that would be good to migrate to a nixos option example while we're doing this cleanup work.

File: plugins/lsp/language-servers/ltex-settings.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/lsp/language-servers/ltex-settings.nix]: Good idea ! Done.

**Review by MattSturgeon — APPROVED** [review]
:broom:

**Comment by GaetanLepage** [comment]
@mergifyio queue

**Pull request merged**